### PR TITLE
docs: add notes on git naming conventions

### PR DESCRIPTION
**Pull request by macintoshpie**
This creates a doc for developer resources which specifies a format for writing commit messages and branch names.

### CODE REVIEW

**Review by markborkum — CHANGES_REQUESTED** [review]
Please include a "proposal" type.

**Review by markborkum — APPROVED** [review]
Looks good, thanks.

**Review by corymosiman12 — APPROVED** [review]

**Review by aviveiros11 — APPROVED** [review]
lgtm 👍

File: docs/developer_resources.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+- `refactor: make SimpleElement restriction a regex test`
+
+Branch examples:
+
+- `1234-feat/typical-operating-hours`
+- `refactor/terminal-unit-type`
+
+Pull request examples:
+
+- `#1234 feat(schema): add MyElement to ParentElement`
```
Comment (nllong) [inline docs/developer_resources.md:45]: 👍

**Review by nllong — APPROVED** [review]
fandidilyastic!

**Pull request merged**